### ISSUE TITLE
libxau doesn't find xproto even if spack finds it

### ISSUE THREAD

**Issue by s-sajid-ali** [spack/spack#8371]
### Expected Result

Successful install

### Actual Result

libxau can't find xproto installed by spack

```
[sajid@xrmlite spack]$ spack install libxau  %intel@18.0.3
==> pkgconf is already installed in /home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/intel-18.0.3/pkgconf-1.4.2-qpzehzoxsht2egj2qaenif3l4tue7koa
==> util-macros is already installed in /home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/intel-18.0.3/util-macros-1.19.1-mcpc6iksxiwi524dnk62ir3v4iobcqiu
==> xproto is already installed in /home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/intel-18.0.3/xproto-7.0.31-6hcwzhe5do4c3mg7wg4kub4laf2un6mj
==> Installing libxau
==> Using cached archive: /home/sajid/packages/spack/var/spack/cache/libxau/libxau-1.0.8.tar.gz
==> Staging archive: /home/sajid/packages/spack/var/spack/stage/libxau-1.0.8-p5ow55z3f3ckzsppuoshkfdqfebcawlm/libXau-1.0.8.tar.gz
==> Created stage in /home/sajid/packages/spack/var/spack/stage/libxau-1.0.8-p5ow55z3f3ckzsppuoshkfdqfebcawlm
==> No patches needed for libxau
==> Building libxau [AutotoolsPackage]
==> Executing phase: 'autoreconf'
==> Executing phase: 'configure'
==> Error: ProcessError: Command exited with status 1:
    '/home/sajid/packages/spack/var/spack/stage/libxau-1.0.8-p5ow55z3f3ckzsppuoshkfdqfebcawlm/libXau-1.0.8/configure' '--prefix=/home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/intel-18.0.3/libxau-1.0.8-p5ow55z3f3ckzsppuoshkfdqfebcawlm'

1 error found in build log:
     124    Perhaps you should add the directory containing `xorg-macros.pc'
     125    to the PKG_CONFIG_PATH environment variable
     126    Package 'xorg-macros', required by 'virtual:world', not found
     127    checking whether make supports nested variables... (cached) yes
     128    checking whether ln -s works... yes
     129    checking for pathconf... yes
  >> 130    checking for XAU... configure: error: Package requirements (xproto) were not met:
     131
     132    Package 'xproto', required by 'virtual:world', not found
     133
     134    Consider adjusting the PKG_CONFIG_PATH environment variable if you
     135    installed software in a non-standard prefix.
     136

See build log for details:
  /home/sajid/packages/spack/var/spack/stage/libxau-1.0.8-p5ow55z3f3ckzsppuoshkfdqfebcawlm/libXau-1.0.8/spack-build.out
```
The pacakge.py for libxau does have xproto though : 
```
    depends_on('xproto', type=('build', 'link'))
    depends_on('pkgconfig', type='build')
    depends_on('util-macros', type='build')
```

[spack-build.out](https://pastebin.com/x46EVMf3)

### Information on your system

 1. linux-centos7-x86_64, intel@18.0.3

**Comment by scheibelp** [comment]
(Note that this may contain sensitive information so you may want to edit it) Could you also include `spack-build.env`?

I'm specifically curious about this because `pkg-config` occasionally omits packages from it's results if variables like `CPATH` are set. This will also include the value of `PKG_CONFIG_PATH` so we can see whether Spack set it properly.

**Comment by s-sajid-ali** [comment]
```
[sajid@xrmlite ~]$ cat /home/sajid/packages/spack/var/spack/stage/libxau-1.0.8-p5ow55z3f3ckzsppuoshkfdqfebcawlm/libXau-1.0.8/spack-build.env
export ACLOCAL_PATH="/home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/intel-18.0.3/pkgconf-1.4.2-qpzehzoxsht2egj2qaenif3l4tue7koa/share/aclocal"
export BASH_FUNC_module()="() {  eval `/usr/bin/modulecmd bash $*`
}"
export BASH_FUNC_spack()="() {  if [ -n "${ZSH_VERSION:-}" ]; then
 emulate -L sh;
 fi;
 args=("$@");
 _sp_flags="";
 while [[ "$1" =~ ^- ]]; do
 _sp_flags="$_sp_flags $1";
 shift;
 done;
 if [[ ( ! -z "$_sp_flags" ) && ( "$_sp_flags" =~ '.*h.*' || "$_sp_flags" =~ '.*V.*' ) ]]; then
 command spack $_sp_flags "$@";
 return;
 fi;
 _sp_subcommand=$1;
 shift;
 _sp_spec=("$@");
 case $_sp_subcommand in
 "cd")
 _sp_arg="$1";
 shift;
 if [ "$_sp_arg" = "-h" ]; then
 command spack cd -h;
 else
 LOC="$(spack location $_sp_arg "$@")";
 if [[ -d "$LOC" ]]; then
 cd "$LOC";
 else
 return 1;
 fi;
 fi;
 return
 ;;
 "use" | "unuse" | "load" | "unload")
 _sp_subcommand_args="";
 _sp_module_args="";
 while [[ "$1" =~ ^- ]]; do
 if [ "$1" = "-r" -o "$1" = "--dependencies" ]; then
 _sp_subcommand_args="$_sp_subcommand_args $1";
 else
 _sp_module_args="$_sp_module_args $1";
 fi;
 shift;
 done;
 _sp_spec=("$@");
 case $_sp_subcommand in
 "use")
 if _sp_full_spec=$(command spack $_sp_flags module loads --input-only $_sp_subcommand_args --module-type dotkit "${_sp_spec[@]}"); then
 use $_sp_module_args $_sp_full_spec;
 fi
 ;;
 "unuse")
 if _sp_full_spec=$(command spack $_sp_flags module loads --input-only $_sp_subcommand_args --module-type dotkit "${_sp_spec[@]}"); then
 unuse $_sp_module_args $_sp_full_spec;
 fi
 ;;
 "load")
 if _sp_full_spec=$(command spack $_sp_flags module loads --input-only $_sp_subcommand_args --module-type tcl "${_sp_spec[@]}"); then
 module load $_sp_module_args $_sp_full_spec;
 fi
 ;;
 "unload")
 if _sp_full_spec=$(command spack $_sp_flags module loads --input-only $_sp_subcommand_args --module-type tcl "${_sp_spec[@]}"); then
 module unload $_sp_module_args $_sp_full_spec;
 fi
 ;;
 esac
 ;;
 *)
 command spack "${args[@]}"
 ;;
 esac
}"
export CC="/home/sajid/packages/spack/lib/spack/env/intel/icc"
export CLASSPATH="/opt/intel/compilers_and_libraries_2018.3.222/linux/mpi/intel64/lib/mpi.jar:/opt/intel/compilers_and_libraries_2018.3.222/linux/daal/lib/daal.jar"
export CMAKE_PREFIX_PATH="/home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/intel-18.0.3/pkgconf-1.4.2-qpzehzoxsht2egj2qaenif3l4tue7koa:/home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/intel-18.0.3/util-macros-1.19.1-mcpc6iksxiwi524dnk62ir3v4iobcqiu:/home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/intel-18.0.3/xproto-7.0.31-6hcwzhe5do4c3mg7wg4kub4laf2un6mj"
export CPATH="/opt/intel/compilers_and_libraries_2018.3.222/linux/ipp/include:/opt/intel/compilers_and_libraries_2018.3.222/linux/mkl/include:/opt/intel/compilers_and_libraries_2018.3.222/linux/pstl/include:/opt/intel/compilers_and_libraries_2018.3.222/linux/tbb/include:/opt/intel/compilers_and_libraries_2018.3.222/linux/daal/include"
export CXX="/home/sajid/packages/spack/lib/spack/env/intel/icpc"
export DAALROOT="/opt/intel/compilers_and_libraries_2018.3.222/linux/daal"
export DK_NODE="/home/sajid/packages/spack/share/spack/dotkit/linux-centos7-x86_64"
export EDITOR="/home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/gcc-8.1.0/neovim-0.2.2-btvzqf4jeacf4vmb637llgcahia4x5jn/bin/nvim"
export F77="/home/sajid/packages/spack/lib/spack/env/intel/ifort"
export F90="/home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/gcc-4.8.5/gcc-7.3.0-xyzezhjmbiebkjfoakso464rhfshlkyq/bin/gfortran"
export FC="/home/sajid/packages/spack/lib/spack/env/intel/ifort"
export GDBSERVER_MIC="/opt/intel/debugger_2018/gdb/targets/intel64/x200/bin/gdbserver"
export GDB_CROSS="/opt/intel/debugger_2018/gdb/intel64/bin/gdb-ia"
export HISTCONTROL="ignoredups"
export HISTSIZE="1000"
export HOME="/home/sajid"
export HOSTNAME="xrmlite.phys.northwestern.edu"
export INFOPATH="/opt/intel/documentation_2018/en/debugger//gdb-ia/info/:/opt/intel/documentation_2018/en/debugger//gdb-igfx/info/"
export INTEL_LICENSE_FILE="/opt/intel/licenses:/opt/intel/compilers_and_libraries_2018.3.222/linux/licenses:/opt/intel/licenses:/home/sajid/intel/licenses"
export INTEL_PYTHONHOME="/opt/intel/debugger_2018/python/intel64/"
export IPPROOT="/opt/intel/compilers_and_libraries_2018.3.222/linux/ipp"
export I_MPI_ROOT="/opt/intel/compilers_and_libraries_2018.3.222/linux/mpi"
export LANG="en_US.UTF-8"
export LC_CTYPE="en_US.UTF-8"
export LD_LIBRARY_PATH="/opt/intel/compilers_and_libraries_2018.3.222/linux/compiler/lib/intel64:/opt/intel/compilers_and_libraries_2018.3.222/linux/mpi/intel64/lib:/opt/intel/compilers_and_libraries_2018.3.222/linux/mpi/mic/lib:/opt/intel/compilers_and_libraries_2018.3.222/linux/ipp/lib/intel64:/opt/intel/compilers_and_libraries_2018.3.222/linux/compiler/lib/intel64_lin:/opt/intel/compilers_and_libraries_2018.3.222/linux/mkl/lib/intel64_lin:/opt/intel/compilers_and_libraries_2018.3.222/linux/tbb/lib/intel64/gcc4.7:/opt/intel/debugger_2018/iga/lib:/opt/intel/debugger_2018/libipt/intel64/lib:/opt/intel/compilers_and_libraries_2018.3.222/linux/daal/lib/intel64_lin:/opt/intel/compilers_and_libraries_2018.3.222/linux/daal/../tbb/lib/intel64_lin/gcc4.4"
export LESSOPEN="||/usr/bin/lesspipe.sh %s"
export LIBRARY_PATH="/opt/intel/compilers_and_libraries_2018.3.222/linux/ipp/lib/intel64:/opt/intel/compilers_and_libraries_2018.3.222/linux/compiler/lib/intel64_lin:/opt/intel/compilers_and_libraries_2018.3.222/linux/mkl/lib/intel64_lin:/opt/intel/compilers_and_libraries_2018.3.222/linux/tbb/lib/intel64/gcc4.7:/opt/intel/compilers_and_libraries_2018.3.222/linux/tbb/lib/intel64/gcc4.7:/opt/intel/compilers_and_libraries_2018.3.222/linux/daal/lib/intel64_lin:/opt/intel/compilers_and_libraries_2018.3.222/linux/daal/../tbb/lib/intel64_lin/gcc4.4"
export LOADEDMODULES="neovim-0.2.2-gcc-8.1.0-btvzqf4:git-2.17.0-gcc-8.1.0-h4rqqlk:gcc-7.3.0-gcc-4.8.5-xyzezhj:intel18"
export LOGNAME="sajid"
export LS_COLORS="rs=0:di=38;5;27:ln=38;5;51:mh=44;38;5;15:pi=40;38;5;11:so=38;5;13:do=38;5;5:bd=48;5;232;38;5;11:cd=48;5;232;38;5;3:or=48;5;232;38;5;9:mi=05;48;5;232;38;5;15:su=48;5;196;38;5;15:sg=48;5;11;38;5;16:ca=48;5;196;38;5;226:tw=48;5;10;38;5;16:ow=48;5;10;38;5;21:st=48;5;21;38;5;15:ex=38;5;34:*.tar=38;5;9:*.tgz=38;5;9:*.arc=38;5;9:*.arj=38;5;9:*.taz=38;5;9:*.lha=38;5;9:*.lz4=38;5;9:*.lzh=38;5;9:*.lzma=38;5;9:*.tlz=38;5;9:*.txz=38;5;9:*.tzo=38;5;9:*.t7z=38;5;9:*.zip=38;5;9:*.z=38;5;9:*.Z=38;5;9:*.dz=38;5;9:*.gz=38;5;9:*.lrz=38;5;9:*.lz=38;5;9:*.lzo=38;5;9:*.xz=38;5;9:*.bz2=38;5;9:*.bz=38;5;9:*.tbz=38;5;9:*.tbz2=38;5;9:*.tz=38;5;9:*.deb=38;5;9:*.rpm=38;5;9:*.jar=38;5;9:*.war=38;5;9:*.ear=38;5;9:*.sar=38;5;9:*.rar=38;5;9:*.alz=38;5;9:*.ace=38;5;9:*.zoo=38;5;9:*.cpio=38;5;9:*.7z=38;5;9:*.rz=38;5;9:*.cab=38;5;9:*.jpg=38;5;13:*.jpeg=38;5;13:*.gif=38;5;13:*.bmp=38;5;13:*.pbm=38;5;13:*.pgm=38;5;13:*.ppm=38;5;13:*.tga=38;5;13:*.xbm=38;5;13:*.xpm=38;5;13:*.tif=38;5;13:*.tiff=38;5;13:*.png=38;5;13:*.svg=38;5;13:*.svgz=38;5;13:*.mng=38;5;13:*.pcx=38;5;13:*.mov=38;5;13:*.mpg=38;5;13:*.mpeg=38;5;13:*.m2v=38;5;13:*.mkv=38;5;13:*.webm=38;5;13:*.ogm=38;5;13:*.mp4=38;5;13:*.m4v=38;5;13:*.mp4v=38;5;13:*.vob=38;5;13:*.qt=38;5;13:*.nuv=38;5;13:*.wmv=38;5;13:*.asf=38;5;13:*.rm=38;5;13:*.rmvb=38;5;13:*.flc=38;5;13:*.avi=38;5;13:*.fli=38;5;13:*.flv=38;5;13:*.gl=38;5;13:*.dl=38;5;13:*.xcf=38;5;13:*.xwd=38;5;13:*.yuv=38;5;13:*.cgm=38;5;13:*.emf=38;5;13:*.axv=38;5;13:*.anx=38;5;13:*.ogv=38;5;13:*.ogx=38;5;13:*.aac=38;5;45:*.au=38;5;45:*.flac=38;5;45:*.mid=38;5;45:*.midi=38;5;45:*.mka=38;5;45:*.mp3=38;5;45:*.mpc=38;5;45:*.ogg=38;5;45:*.ra=38;5;45:*.wav=38;5;45:*.axa=38;5;45:*.oga=38;5;45:*.spx=38;5;45:*.xspf=38;5;45:"export MAIL="/var/spool/mail/sajid"
export MANPATH="/opt/intel/man/common:/opt/intel/compilers_and_libraries_2018.3.222/linux/mpi/man:/opt/intel/documentation_2018/en/debugger//gdb-ia/man/:/opt/intel/documentation_2018/en/debugger//gdb-igfx/man/:/home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/gcc-4.8.5/gcc-7.3.0-xyzezhjmbiebkjfoakso464rhfshlkyq/share/man:/home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/gcc-8.1.0/git-2.17.0-h4rqqlkzwe3zjebheonbftftq5jm7pig/share/man:/home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/gcc-8.1.0/neovim-0.2.2-btvzqf4jeacf4vmb637llgcahia4x5jn/share/man:/usr/share/man:/usr/local/share/man:"
export MKLROOT="/opt/intel/compilers_and_libraries_2018.3.222/linux/mkl"
export MODULEPATH="/home/sajid/packages/spack/share/spack/modules/linux-centos7-x86_64:/usr/share/Modules/modulefiles:/etc/modulefiles"
export MODULESHOME="/usr/share/Modules"
export MPM_LAUNCHER="/opt/intel/debugger_2018/mpm/mic/bin/start_mpm.sh"
export NLSPATH="/opt/intel/compilers_and_libraries_2018.3.222/linux/compiler/lib/intel64/locale/%l_%t/%N:/opt/intel/compilers_and_libraries_2018.3.222/linux/mkl/lib/intel64_lin/locale/%l_%t/%N:/opt/intel/debugger_2018/gdb/intel64/share/locale/%l_%t/%N"
export PATH="/home/sajid/packages/spack/lib/spack/env:/home/sajid/packages/spack/lib/spack/env/case-insensitive:/home/sajid/packages/spack/lib/spack/env/intel:/home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/intel-18.0.3/pkgconf-1.4.2-qpzehzoxsht2egj2qaenif3l4tue7koa/bin:/home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/intel-18.0.3/pkgconf-1.4.2-qpzehzoxsht2egj2qaenif3l4tue7koa/bin:/opt/intel/compilers_and_libraries_2018.3.222/linux/bin/intel64:/opt/intel/compilers_and_libraries_2018.3.222/linux/mpi/intel64/bin:/home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/gcc-4.8.5/gcc-7.3.0-xyzezhjmbiebkjfoakso464rhfshlkyq/bin:/home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/gcc-8.1.0/git-2.17.0-h4rqqlkzwe3zjebheonbftftq5jm7pig/bin:/home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/gcc-8.1.0/neovim-0.2.2-btvzqf4jeacf4vmb637llgcahia4x5jn/bin:/home/sajid/packages/spack/bin:/usr/lib64/qt-3.3/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/home/sajid/.local/bin:/home/sajid/bin:/opt/intel/parallel_studio_xe_2018.3.051/bin"
export PKG_CONFIG_PATH="/opt/intel/compilers_and_libraries_2018.3.222/linux/mkl/bin/pkgconfig"
export PSTLROOT="/opt/intel/compilers_and_libraries_2018.3.222/linux/pstl"
export PWD="/home/sajid"
export QTDIR="/usr/lib64/qt-3.3"
export QTINC="/usr/lib64/qt-3.3/include"
export QTLIB="/usr/lib64/qt-3.3/lib"
export QT_GRAPHICSSYSTEM_CHECKED="1"
export SELINUX_LEVEL_REQUESTED=""
export SELINUX_ROLE_REQUESTED=""
export SELINUX_USE_CURRENT_RANGE=""
export SHELL="/bin/bash"
export SHLVL="1"
export SPACK_CC="/opt/intel/compilers_and_libraries_2018.3.222/linux/bin/intel64/icc"
export SPACK_CC_RPATH_ARG="-Wl,-rpath,"
export SPACK_COMPILER_EXTRA_RPATHS="/opt/intel/compilers_and_libraries_2018.3.222/linux/compiler/lib/intel64_lin"
export SPACK_COMPILER_SPEC="intel@18.0.3"
export SPACK_CXX="/opt/intel/compilers_and_libraries_2018.3.222/linux/bin/intel64/icpc"
export SPACK_CXX_RPATH_ARG="-Wl,-rpath,"
export SPACK_DEBUG_LOG_DIR="/home/sajid"
export SPACK_DEBUG_LOG_ID="libxau-p5ow55z"
export SPACK_DEPENDENCIES="/home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/intel-18.0.3/pkgconf-1.4.2-qpzehzoxsht2egj2qaenif3l4tue7koa:/home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/intel-18.0.3/util-macros-1.19.1-mcpc6iksxiwi524dnk62ir3v4iobcqiu:/home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/intel-18.0.3/xproto-7.0.31-6hcwzhe5do4c3mg7wg4kub4laf2un6mj"
export SPACK_ENV_PATH="/home/sajid/packages/spack/lib/spack/env:/home/sajid/packages/spack/lib/spack/env/case-insensitive:/home/sajid/packages/spack/lib/spack/env/intel"
export SPACK_F77="/opt/intel/compilers_and_libraries_2018.3.222/linux/bin/intel64/ifort"
export SPACK_F77_RPATH_ARG="-Wl,-rpath,"
export SPACK_FC="/opt/intel/compilers_and_libraries_2018.3.222/linux/bin/intel64/ifort"
export SPACK_FC_RPATH_ARG="-Wl,-rpath,"
export SPACK_INSTALL="/home/sajid/packages/spack/opt/spack"
export SPACK_LINK_DEPS="/home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/intel-18.0.3/xproto-7.0.31-6hcwzhe5do4c3mg7wg4kub4laf2un6mj"
export SPACK_PREFIX="/home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/intel-18.0.3/libxau-1.0.8-p5ow55z3f3ckzsppuoshkfdqfebcawlm"
export SPACK_ROOT="/home/sajid/packages/spack"
export SPACK_RPATH_DEPS="/home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/intel-18.0.3/xproto-7.0.31-6hcwzhe5do4c3mg7wg4kub4laf2un6mj"
export SPACK_SHELL="bash"
export SPACK_SHORT_SPEC="libxau@1.0.8%intel@18.0.3 arch=linux-centos7-x86_64 /p5ow55z"
export SSH_CLIENT="Redacted"
export SSH_CONNECTION="Redacted"
export SSH_TTY="/dev/pts/1"
export TBBROOT="/opt/intel/compilers_and_libraries_2018.3.222/linux/tbb"
export TERM="xterm-256color"
export USER="sajid"
export XDG_DATA_DIRS="/home/sajid/.local/share/flatpak/exports/share/:/var/lib/flatpak/exports/share/:/usr/local/share/:/usr/share/"
export XDG_RUNTIME_DIR="/run/user/1003"
export XDG_SESSION_ID="828"
export _="/home/sajid/packages/spack/bin/spack"
export _LMFILES_="/home/sajid/packages/spack/share/spack/modules/linux-centos7-x86_64/neovim-0.2.2-gcc-8.1.0-btvzqf4:/home/sajid/packages/spack/share/spack/modules/linux-centos7-x86_64/git-2.17.0-gcc-8.1.0-h4rqqlk:/home/sajid/packages/spack/share/spack/modules/linux-centos7-x86_64/gcc-7.3.0-gcc-4.8.5-xyzezhj:/home/sajid/packages/spack/share/spack/modules/linux-centos7-x86_64/intel18"
```

**Comment by s-sajid-ali** [comment]
This does not work either : 
```
[sajid@xrmlite ~]$ export PKG_CONFIG_PATH=/home/sajid/packages/spack/opt/spack/linux-centos7-x86_64/intel-18.0.3/xproto-7.0.31-6hcwzhe5do4c3mg7wg4kub4laf2un6mj/
[sajid@xrmlite ~]$ spack install --dirty libxau  %intel@18.0.3
```

**Comment by scheibelp** [comment]
I'm assuming there is a module associated with your intel compiler? If so, does it modify `PKG_CONFIG_PATH`? If so, could you post the relevant modifications from the module file?

If all the above is true, then this is another case of https://github.com/spack/spack/issues/8345, and it also explains why `--dirty` doesn't help (because the module is overriding `PKG_CONFIG_PATH`). The fix should take a few days.

**Comment by s-sajid-ali** [comment]
I created the module myself and because I could not get spack to install the intel compilers. Below is the module file that I'm using : 
```
[sajid@xrmlite linux-centos7-x86_64]$ cat intel18
#%Module
setenv I_MPI_ROOT {/opt/intel/compilers_and_libraries_2018.3.222/linux/mpi};
setenv GDB_CROSS {/opt/intel/debugger_2018/gdb/intel64/bin/gdb-ia};
setenv INTEL_PYTHONHOME {/opt/intel/debugger_2018/python/intel64/};
setenv IPPROOT {/opt/intel/compilers_and_libraries_2018.3.222/linux/ipp};
setenv DAALROOT {/opt/intel/compilers_and_libraries_2018.3.222/linux/daal};
prepend-path PATH {/opt/intel/compilers_and_libraries_2018.3.222/linux/mpi/intel64/bin};
prepend-path PATH {/opt/intel/compilers_and_libraries_2018.3.222/linux/bin/intel64};
append-path PATH {/opt/intel/parallel_studio_xe_2018.3.051/bin};
setenv INTEL_LICENSE_FILE {/opt/intel/licenses:/opt/intel/compilers_and_libraries_2018.3.222/linux/licenses:/opt/intel/licenses:/home/sajid/intel/licenses};
prepend-path CPATH {/opt/intel/compilers_and_libraries_2018.3.222/linux/daal/include};
prepend-path CPATH {/opt/intel/compilers_and_libraries_2018.3.222/linux/tbb/include};
prepend-path CPATH {/opt/intel/compilers_and_libraries_2018.3.222/linux/tbb/include};
prepend-path CPATH {/opt/intel/compilers_and_libraries_2018.3.222/linux/pstl/include};
prepend-path CPATH {/opt/intel/compilers_and_libraries_2018.3.222/linux/mkl/include};
prepend-path CPATH {/opt/intel/compilers_and_libraries_2018.3.222/linux/ipp/include};
prepend-path LD_LIBRARY_PATH {/opt/intel/compilers_and_libraries_2018.3.222/linux/daal/../tbb/lib/intel64_lin/gcc4.4};
prepend-path LD_LIBRARY_PATH {/opt/intel/compilers_and_libraries_2018.3.222/linux/daal/lib/intel64_lin};
prepend-path LD_LIBRARY_PATH {/opt/intel/debugger_2018/libipt/intel64/lib};
prepend-path LD_LIBRARY_PATH {/opt/intel/debugger_2018/iga/lib};
prepend-path LD_LIBRARY_PATH {/opt/intel/compilers_and_libraries_2018.3.222/linux/tbb/lib/intel64/gcc4.7};
prepend-path LD_LIBRARY_PATH {/opt/intel/compilers_and_libraries_2018.3.222/linux/tbb/lib/intel64/gcc4.7};
prepend-path LD_LIBRARY_PATH {/opt/intel/compilers_and_libraries_2018.3.222/linux/mkl/lib/intel64_lin};
prepend-path LD_LIBRARY_PATH {/opt/intel/compilers_and_libraries_2018.3.222/linux/compiler/lib/intel64_lin};
prepend-path LD_LIBRARY_PATH {/opt/intel/compilers_and_libraries_2018.3.222/linux/ipp/lib/intel64};
prepend-path LD_LIBRARY_PATH {/opt/intel/compilers_and_libraries_2018.3.222/linux/mpi/mic/lib};
prepend-path LD_LIBRARY_PATH {/opt/intel/compilers_and_libraries_2018.3.222/linux/mpi/intel64/lib};
prepend-path LD_LIBRARY_PATH {/opt/intel/compilers_and_libraries_2018.3.222/linux/compiler/lib/intel64_lin};
prepend-path LD_LIBRARY_PATH {/opt/intel/compilers_and_libraries_2018.3.222/linux/compiler/lib/intel64};
prepend-path MANPATH {/opt/intel/documentation_2018/en/debugger//gdb-igfx/man/};
prepend-path MANPATH {/opt/intel/documentation_2018/en/debugger//gdb-ia/man/};
prepend-path MANPATH {/opt/intel/compilers_and_libraries_2018.3.222/linux/mpi/man};
prepend-path MANPATH {/opt/intel/man/common};
append-path MANPATH {};
setenv MPM_LAUNCHER {/opt/intel/debugger_2018/mpm/mic/bin/start_mpm.sh};
setenv PKG_CONFIG_PATH {/opt/intel/compilers_and_libraries_2018.3.222/linux/mkl/bin/pkgconfig};
setenv LIBRARY_PATH {/opt/intel/compilers_and_libraries_2018.3.222/linux/ipp/lib/intel64:/opt/intel/compilers_and_libraries_2018.3.222/linux/compiler/lib/intel64_lin:/opt/intel/compilers_and_libraries_2018.3.222/linux/mkl/lib/intel64_lin:/opt/intel/compilers_and_libraries_2018.3.222/linux/tbb/lib/intel64/gcc4.7:/opt/intel/compilers_and_libraries_2018.3.222/linux/tbb/lib/intel64/gcc4.7:/opt/intel/compilers_and_libraries_2018.3.222/linux/daal/lib/intel64_lin:/opt/intel/compilers_and_libraries_2018.3.222/linux/daal/../tbb/lib/intel64_lin/gcc4.4};
setenv TBBROOT {/opt/intel/compilers_and_libraries_2018.3.222/linux/tbb};
setenv INFOPATH {/opt/intel/documentation_2018/en/debugger//gdb-ia/info/:/opt/intel/documentation_2018/en/debugger//gdb-igfx/info/};
setenv GDBSERVER_MIC {/opt/intel/debugger_2018/gdb/targets/intel64/x200/bin/gdbserver};
setenv MKLROOT {/opt/intel/compilers_and_libraries_2018.3.222/linux/mkl};
setenv NLSPATH {/opt/intel/compilers_and_libraries_2018.3.222/linux/compiler/lib/intel64/locale/%l_%t/%N:/opt/intel/compilers_and_libraries_2018.3.222/linux/mkl/lib/intel64_lin/locale/%l_%t/%N:/opt/intel/debugger_2018/gdb/intel64/share/locale/%l_%t/%N};
setenv CLASSPATH {/opt/intel/compilers_and_libraries_2018.3.222/linux/mpi/intel64/lib/mpi.jar:/opt/intel/compilers_and_libraries_2018.3.222/linux/daal/lib/daal.jar};
setenv PSTLROOT {/opt/intel/compilers_and_libraries_2018.3.222/linux/pstl};
```

This might also explain why I end up with [8356](https://github.com/spack/spack/issues/8356)

**Comment by scheibelp** [comment]
> I created the module myself and because I could not get spack to install the intel compilers

Is that in an issue somewhere? Feel free to ping on it if it hasn't gotten attention lately.

The issue here appears to be:

`setenv PKG_CONFIG_PATH ...`

Can you replace that with:

`prepend-path PKG_CONFIG_PATH ...`

?

**Comment by scheibelp** [comment]
I suppose I'm saying that this hasn't come up before and so if you have full control over the module file then perhaps editing it would the the shortest path to resolving the issue.

**Comment by s-sajid-ali** [comment]
That change made it work.

**Comment by scheibelp** [comment]
Great! I'm going to close this since I think this is a special case. Feel free to reopen if you disagree.

**Comment by s-sajid-ali** [comment]
My group purchased intel compilers for the workstation and we got a floating license and I had to first install the license manager for it. What spack never told me was that this license manager was missing some dependencies and since this was not a spack package, I had to install it using the link given by intel. 
I posted the issue [here](https://github.com/spack/spack/pull/7469) and asked in the slack group and the intel support team who said that they don't support spack package manager but tried to help.

Since this package was installed at ```/opt/intel/ ```, the compilers were also getting installed there. After two days of trying many different ways to change this, I installed them without spack and used them as externally defined compilers.
